concordium-smart-contract-testing/src/lib.rs

```diff
@@ -0,0 +1,2745 @@
+                                },
+                                None => InvokeResponse::Failure {
+                                    kind: v1::InvokeFailure::NonExistentContract,
+                                },
+                            };
+
+                            let energy_after_invoke =
+                                remaining_energy - QUERY_CONTRACT_BALANCE_COST;
+
+                            let resume_res = v1::resume_receive(
+                                config,
+                                response,
+                                InterpreterEnergy::from(energy_after_invoke),
+                                &mut self.mutable_state,
+                                false, // State never changes on queries.
+                                self.loader,
+                            );
+
+                            self.process(resume_res)
+                        }
+                        v1::Interrupt::QueryExchangeRates => {
+                            println!("Querying exchange rates");
+
+                            let exchange_rates =
+                                (self.chain.euro_per_energy, self.chain.micro_ccd_per_euro);
+
+                            let response = InvokeResponse::Success {
+                                new_balance: self.chain.get_instance(self.address)?.self_balance,
+                                data:        Some(to_bytes(&exchange_rates)),
+                            };
+
+                            let energy_after_invoke = remaining_energy - QUERY_EXCHANGE_RATE_COST;
+
+                            let resume_res = v1::resume_receive(
+                                config,
+                                response,
+                                InterpreterEnergy::from(energy_after_invoke),
+                                &mut self.mutable_state,
+                                false, // State never changes on queries.
+                                self.loader,
+                            );
+
+                            self.process(resume_res)
+                        }
+                    }
+                }
+                x => Ok(x),
+            },
+            Err(e) => Err(e),
+        }
+    }
+}
+
+#[derive(Debug, Error)]
+#[error("Module has not been deployed.")]
+struct ModuleMissing;
+
+impl From<ModuleMissing> for ContractInitError {
+    fn from(_: ModuleMissing) -> Self { Self::ModuleDoesNotExist }
+}
+
+impl From<ModuleMissing> for ContractUpdateError {
+    fn from(_: ModuleMissing) -> Self { Self::ModuleDoesNotExist }
+}
+
+#[derive(Debug, Error)]
+#[error("Contract instance has not been instantiated.")]
+struct ContractInstanceMissing;
+
+impl From<ContractInstanceMissing> for ContractUpdateError {
+    fn from(_: ContractInstanceMissing) -> Self { Self::InstanceDoesNotExist }
+}
+
+#[derive(Debug, Error)]
+#[error("Account has not been created.")]
+struct AccountMissing;
```

Comment:
Would it not make sense to include the account that is missing in the error? So
```rust
struct AccountMissing{
    missing_account: AccountAddress
}
```

Reply:
Yes, added for module missing and contract missing as well.